src/cryptonote_core/service_node_voting.h

```diff
@@ -148,6 +187,11 @@ namespace service_nodes
     std::vector<checkpoint_pool_entry> m_checkpoint_pool;
 
     mutable epee::critical_section m_lock;
+
+    using deregister_pool_array       = std::array<std::vector<deregister_pool_entry>       *, 2>;
+    using deregister_pool_array_const = std::array<std::vector<deregister_pool_entry> const *, 2>;
+    deregister_pool_array       get_deregister_pools      ();
+    deregister_pool_array_const get_deregister_pools_const() const;
```

Comment:
Why `_const` for the const version instead of just offering const and non-const versions?

Reply:
My reason is simple. I avoid overloading a lot. But the double const is a good point and I agree.